rest-api/rest-api.php

```diff
@@ -0,0 +1,672 @@
+
+
+        $this->update_maarifa_contact( $request, $id_upd, $fields3 );
+    }
+
+
+    public function update_maarifa_contact( WP_REST_Request $request, int $p_post_id, $fields2 ){
+
+
+        dt_write_log( 'Update_maarifa_contact' );
+
+        if ( isset( $p_post_id ) )
+        {
+
+            $post_id = $p_post_id;
+        }
+        else {
+            $post_id = $url_params['id'];
+        }
+
+        $url_params = $request->get_url_params();
+        $get_params = $request->get_query_params();
+        $silent = isset( $get_params['silent'] ) && $get_params['silent'] === 'true';
+
+        return DT_Posts::update_post( $url_params['post_type'], $post_id, $fields2, $silent );
+    }
+
+
+    public function check_field_value_exists( WP_REST_Request $request, string $maarifa_contact_id ) {
+        //Verify if maarifa_data already exists
+
+        dt_write_log( 'Check_field_value_exists' );
+
+        $params = $request->get_params();
+
+        if ( $maarifa_contact_id != null ) {
+
+            global $wpdb;
+            $result = $wpdb->get_results( $wpdb->prepare(
+                "SELECT `post_id`
+                            FROM $wpdb->postmeta
+                            WHERE meta_key LIKE `maarifa_data`
+                            AND meta_value = %s;", $maarifa_contact_id ) );
+
+
+            return $result;
+
+        }
+        return [];
+    }
+
+
+    public function add_interactions( WP_REST_Request $request ){
+
+        dt_write_log( 'Add_interactions' );
+
+        $url_params = $request->get_url_params();
+        $get_params = $request->get_query_params();
+        $body = $request->get_json_params() ?? $request->get_body_params();
+        $silent = isset( $get_params['silent'] ) && $get_params['silent'] === 'true';
+        $args = [];
+
+        $type = 'maarifa';
+
+        if ( !empty( $body ) ) {
+
+            $value = null;
+            $result = null;
+            $ret = null;
+
+            foreach ( $body as $key => $value ) {
+
+
+                if ( isset( $value['when_made'] ) ){
+                    $args['comment_date'] = $value['when_made'];
+                }
+
+                if ( isset( $value['responder_name'] ) ){
+                    $args['user_id'] = $value['responder_name'];
+                }
+
+                if ( isset( $value['notes'] ) ){
+
+                    $comment = 'Type: '. $type ."\n". 'Responder name: '. $value['responder_name']. "\n" .$value['notes'];
+
+                }
+
+                if ( isset( $value['meta'] ) ) {
+                    $args['comment_meta'] = $value['meta'];
+                }
+
+                if ( isset( $value['id'] ) )
+                {//If comment_id exists, update the comment
+
+                    $id = $value['id'];
+
+                    dt_write_log( 'Add_interactions UPDATE' );
+                    dt_write_log( $id );
+
+                    $result = DT_Posts::update_post_comment( $id, $comment, true, $type, $args );
```

Comment:
The `id` that is passed in the JSON data is not the id of the comment in DT. It is the ID of the interaction in the RS. So we can't use that to update an existing comment. We could probably use that id if we save it as extra meta data on the comment, but that is a little more complicated and we can handle that scenario later. For now, let's just always add new comments.